paymentsheet/src/main/java/com/stripe/android/link/ui/wallet/WalletScreen.kt

```diff
@@ -112,50 +115,54 @@ internal fun WalletBody(
     showBottomSheetContent: (BottomSheetContent) -> Unit,
     hideBottomSheetContent: () -> Unit
 ) {
-    if (state.paymentDetailsList.isEmpty()) {
-        Loader()
-        return
-    }
-
-    if (state.alertMessage != null) {
-        AlertMessage(
-            alertMessage = state.alertMessage,
-            onDismissAlert = onDismissAlert
-        )
-    }
+    AnimatedContent(
+        targetState = state.paymentDetailsList.isEmpty(),
+        transitionSpec = { LinkScreenTransition },
+    ) { isLoading ->
+        if (isLoading) {
+            LinkLoadingScreen(Modifier.testTag(WALLET_LOADER_TAG))
+        } else {
```

Comment:
👀 `AnimatedContent` smooths out the transition. Demo of the problem:

https://github.com/user-attachments/assets/cfcc73db-b360-43d2-b837-ed4bd12d5615